packages/calcite-components/src/components/input-number/input-number.e2e.ts

```diff
@@ -1300,6 +1300,57 @@ describe("calcite-input-number", () => {
           const input = await page.find("calcite-input-number >>> input");
           expect(await input.getProperty("value")).toBe("0");
         });
+
+        it(`allows editing numbers that start with zeros ${locale} locale`, async () => {
+          const page = await newE2EPage();
+          await page.setContent(html`<calcite-input-number locale=${locale}></calcite-input-number>`);
+
+          const calciteInput = await page.find("calcite-input-number");
+          const input = await page.find("calcite-input-number >>> input");
+          await calciteInput.callMethod("setFocus");
+          await page.waitForChanges();
+          await typeNumberValue(page, "7000");
+          await page.waitForChanges();
+          expect(await calciteInput.getProperty("value")).toBe("7000");
+          expect(await input.getProperty("value")).toBe("7000");
+
+          await page.keyboard.press("ArrowLeft");
+          await page.keyboard.press("ArrowLeft");
+          await page.keyboard.press("ArrowLeft");
+          await page.keyboard.press("Backspace");
+          await page.waitForChanges();
+          expect(await calciteInput.getProperty("value")).toBe("0");
+          expect(await input.getProperty("value")).toBe("000");
+
+          await page.keyboard.type("5");
+          await page.waitForChanges();
+          expect(await calciteInput.getProperty("value")).toBe("5000");
+          expect(await input.getProperty("value")).toBe("5000");
+
+          await page.keyboard.press("Backspace");
+          await page.keyboard.press("Delete");
+          await page.keyboard.press("Delete");
+          await page.keyboard.press("Delete");
+          await page.waitForChanges();
+          await typeNumberValue(page, "70011");
+          await page.waitForChanges();
+          expect(await calciteInput.getProperty("value")).toBe("70011");
+          expect(await input.getProperty("value")).toBe("70011");
+
+          await page.keyboard.press("ArrowLeft");
+          await page.keyboard.press("ArrowLeft");
+          await page.keyboard.press("ArrowLeft");
+          await page.keyboard.press("ArrowLeft");
+          await page.keyboard.press("Backspace");
+          await page.waitForChanges();
+          expect(await calciteInput.getProperty("value")).toBe("11");
+          expect(await input.getProperty("value")).toBe("0011");
+
+          await page.keyboard.type("5");
+          await page.waitForChanges();
+          expect(await calciteInput.getProperty("value")).toBe("50011");
+          expect(await input.getProperty("value")).toBe("50011");
+        });
```

Comment:
I don't think we need to test this feature for every locale. 

Can you remove this test and add a new one on line 1544 that does the following:

- uses the `arab` numbering system 
- uses the `ar` locale
- uses a minus sign
- uses a decimal